core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -611,7 +613,8 @@ public void restart(String workflowId, boolean useLatestDefinitions) {
             throw e;
         }
 
-        decide(workflowId);
+        metadataMapperService.populateWorkflowWithDefinitions(workflow);
+        decide(workflow.getWorkflowId());
```

Comment:
can `workflow` be passed in here instead of just workflowId, so it won't need to reload the workflow model again?

Reply:
keeping it consistent with other re* methods like re-run, retry etc to always re-load frrom the database and evaluate.  Given these are not very frequently used methods.